web/experiments/feature_view.py

```diff
@@ -0,0 +1,240 @@
+import datetime
+import logging
+import os
+import re
+import sys
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from matplotlib import pyplot as plt
+from sklearn.linear_model import SGDClassifier
+from sklearn.metrics import f1_score, make_scorer
+from sklearn.model_selection import LeaveOneOut, GridSearchCV
+from tqdm import tqdm
+
+
+def process_CSLB(*embs, feature_matrix_path='../CSLB/feature_matrix.dat'):
+    df = pd.read_csv(feature_matrix_path, sep='\t', index_col=0)
+
+    t = df.transpose()
+    t.iloc[t.values > 1] = 1
+
+    iv = df.index.values.astype(str)
+
+    word_to_rm = set([])
+
+    # filter double meaning words
+    for cw in iv:
+        sw = str(cw)
+
+        ssw = sw.split('_(')
+        if len(ssw) >= 2:
+            word_to_rm.add(sw)
+        else:
+            # remove two-word names according to paper
+            # 'guinea_pig', 'rolls_royce'
+            ssw = sw.split('_')
+            if len(ssw) >= 2:
+                word_to_rm.add(sw)
+
+    word_to_rm = [str(x) for x in word_to_rm]
+
+    # drop words non aviable in embeddings corpora
+    concepts_str = [str(x) for x in t.columns]
+
+    def get_unexisting_words(emb):
+        lack_words = []
+
+        for e in emb:
+            for c in concepts_str:
+                if c not in e.words:
+                    lack_words.append(c)
+
+        return set(lack_words)
+
+    uw = get_unexisting_words(embs)
+    word_to_rm.extend(uw)
+
+    # remove before feature selection
+    cleaned_df = t.drop(word_to_rm, axis=1)
+
+    f_summed = cleaned_df.sum(axis=1)
+
+    # feature with 5 or more positive examples
+    selected_feature = f_summed.loc[f_summed >= 5]
+
+    # DataFrame with removed absent word in embedding and
+    cleaned_df = cleaned_df.loc[selected_feature.index.values.astype(str)]
+
+    return cleaned_df
+
+
+def learn_logit_reg(embedding, features, concepts, cleaned_norms, n_jobs=4, random_state=None, nb_hyper=20,
```

Comment:
Internal functions (i.e. ones not visible to the user) should be prefixed with a "_"